torch/_inductor/package/package.py

```diff
@@ -0,0 +1,214 @@
+import glob
+import io
+import json
+import os
+import re
+import shlex
+import subprocess
+import tempfile
+import zipfile
+from typing import Any, Callable, Dict, List, Union
+
+import torch
+import torch._inductor
+import torch.utils._pytree as pytree
+from torch._inductor import config, exc
+from torch._utils_internal import get_file_path_2
+from torch.export._tree_utils import reorder_kwargs
+
+
+ARCHIVE_VERSION = 0
+
+with open(
+    get_file_path_2(os.path.dirname(__file__), "pt2_archive_constants.json")
+) as f:
+    PT2_ARCHIVE_CONSTANTS = json.load(f)
+
+AOTINDUCTOR_DIR = PT2_ARCHIVE_CONSTANTS["AOTINDUCTOR_DIR"]
+CONSTANTS_DIR = PT2_ARCHIVE_CONSTANTS["CONSTANTS_DIR"]
+
+
+class PT2ArchiveWriter:
+    def __init__(self, archive_path: str):
+        self.archive_file = zipfile.ZipFile(archive_path, "w")
+        self.write_string("version", str(ARCHIVE_VERSION))
+        self.write_string("archive_format", "pt2")
```

Comment:
Shouldn't logic from `__init__` be moved to the enter? (and exit should do `self.archive_file.close`?)